riptable/rt_str.py

```diff
@@ -713,8 +715,105 @@ def regex_match(self, regex):
         regex = re.compile(regex)
         vmatch = np.vectorize(lambda x: bool(regex.search(x)))
         bools = vmatch(self.backtostring)
+
         return bools
 
+    def extract(self, regex: str, expand: Optional[bool] = None,
+                fillna: str = '', names=None, apply_unique: bool = True
+                ) -> Union[FastArray, TypeRegister.Dataset]:

```

Comment:
One small thing here -- you don't want to import ``Dataset`` above (in the normal way), but ``TypeRegister.Dataset`` won't really be understood by type-checkers like mypy. The best way to do this is to use the special ``typing.TYPE_CHECKING`` constant and do a conditional import of Dataset like so:
```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from .rt_dataset import Dataset
```

Then, you can use it here by specifying the ``Dataset`` name as a string; so the return type annotation for ``extract`` will be:
```
Union[FastArray, 'Dataset']
```